### ISSUE TITLE
Reducing size of labels produced on detect.py

### ISSUE THREAD

**Issue by mattryanong** [ultralytics/yolov5#6935]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

How can i reduce the size of the labels that are drawn with the bboxes so that i can display each label with its associated confidence clearly in the image?

![11](https://user-images.githubusercontent.com/54696024/157677493-777ce70a-a87f-4198-b311-a9eaee4249ca.jpg)

For example, I would like to see the values for detecting each of these stripes in a glance as they are currently overlapping each other and masking each other.

**Comment by glenn-jocher** [comment]
@mattryanong you can run `--hide-conf` or `--hide-label` to remove these:
https://github.com/ultralytics/yolov5/blob/055e72af5b887832d5e7267ac9226c825d498cd2/detect.py#L244-L245

**Comment by mattryanong** [comment]
is it possible to reduce the font size of the labels- so that i can retain the labels and confs but just have them smaller instead of removing them altogether?

**Comment by glenn-jocher** [comment]
@mattryanong yes, you can try to customize the YOLOv5 Annotator:
https://github.com/ultralytics/yolov5/blob/055e72af5b887832d5e7267ac9226c825d498cd2/utils/plots.py#L65-L119

detect.py creates an Annotator instance here:
https://github.com/ultralytics/yolov5/blob/055e72af5b887832d5e7267ac9226c825d498cd2/detect.py#L157

**Comment by mattryanong** [comment]
This helps so much thank you!!!

**Comment by shashankrajput-iitk** [comment]
@mattryanong What changes do you made to reduce the font size of label? Do you mind sharing the changes in the Annotator class or your plots.py file? It will be very kind of you for the help.

**Comment by YoniME** [comment]
> @mattryanong you can run `--hide-conf` or `--hide-label` to remove these:
> 
> https://github.com/ultralytics/yolov5/blob/055e72af5b887832d5e7267ac9226c825d498cd2/detect.py#L244-L245

Hi @glenn-jocher,

I'm running the model with torch, and loading it with `torch.hub.load` .
What is the best way of hiding the label / reducing the thickness in that case?

Many thanks!

**Comment by glenn-jocher** [comment]
@YoniME you can explore AutoShape() and Detections() class methods here:
https://github.com/ultralytics/yolov5/blob/541a5b72bbb78b4eac8ef9961daa7b06399b8c2b/models/common.py#L517-L716

**Comment by YoniME** [comment]
@glenn-jocher Thanks!

**Comment by PawelKinczyk** [comment]
Hi, i try to change the line weight and font size but i don't know where to start. My code looks like this:
```
import cv2
import torch
from mss import mss
import numpy as np
from PIL import Image


model = torch.hub.load("yolov5", 'custom', path="train_results/traine_best/weights/best.pt", source='local')
model.conf = 0.6
sct = mss()

while 1:
    w, h = 1920, 1080
    monitor = {'top': 0, 'left': 0, 'width': w, 'height': h}
    img = Image.frombytes('RGB', (w, h), sct.grab(monitor).rgb)
    screen = cv2.cvtColor(np.array(img), cv2.COLOR_RGB2BGR)
    # set the model use the screen

    result = model(screen, size=1280)
    result.line_thickness = 0.1  # Change line weight of bounding boxes
    result.font_size = 0.1  # Change font size of labels
    cv2.imshow('Screen', result.render()[0])

    if cv2.waitKey(25) & 0xFF == ord('q'):
        cv2.destroyAllWindows()
        break
```
pleas help me to achieve this

**Comment by glenn-jocher** [comment]
@PawelKinczyk hi,

To change the line weight and font size in YOLOv5, you can modify the `line_thickness` and `font_size` attributes of the `result` object. 

In your code, you can set the `line_thickness` and `font_size` before displaying the image. Here's an example:

```python
result.line_thickness = 2  # Change line weight of bounding boxes
result.font_size = 20  # Change font size of labels
cv2.imshow('Screen', result.render()[0])
```

By adjusting the values of `line_thickness` and `font_size`, you can control the thickness of the bounding box lines and the size of the labels, respectively.

I hope this helps! Let me know if you have any further questions.

**Comment by PawelKinczyk** [comment]
It doesn't work... :(
    result.line_thickness = 1  # Change line weight of bounding boxes
    result.font_size = 1  # Change font size of labels
![image](https://github.com/ultralytics/yolov5/assets/96824698/641480cb-a59d-40fe-93a8-944e7a9698fa)
and
    result.line_thickness = 100  # Change line weight of bounding boxes
    result.font_size = 100  # Change font size of labels
![image](https://github.com/ultralytics/yolov5/assets/96824698/f12b4b91-8a89-4603-8a99-4dfe0002b6f0)

**Comment by oliwhi** [comment]
Why is this issue closed? This is still a persisting problem. The `font_size` argument seems to get lost when defined in the the `annotator` object is called. Same with line thickness. It seems the the font-size and line thickness variables are overridden as a function of the image size. Did I overlook something or is this still a problem for anyone else?

**Comment by glenn-jocher** [comment]
@oliwhi the issue was closed because the `font_size` and `line_thickness` attributes in YOLOv5 are not intended to control the font size and line thickness directly. They are used internally to scale the size of the annotations based on the image size. 

If you want to adjust the font size and line thickness of the annotations in YOLOv5, you can modify the `Annotator` class in the `plots.py` file of the YOLOv5 repository. Specifically, you can modify the `plot_one_box` function to change the font size and line thickness of the annotations.

Please note that modifying the source code is outside the scope of support we can provide here. However, you can refer to the `Annotator` class and the `plot_one_box` function in the YOLOv5 repository to implement the desired changes.

If you have any further questions or need any assistance, feel free to ask.

**Comment by Sscicada** [comment]
@PawelKinczyk I'm having the same issue, and my workaround is like this: modify the _run function of the Detections class in the common.py file in the models directory. Probably on line 906, specify the size of the line_width parameter.

```python
if show or save or render or crop:
    annotator = Annotator(im, line_width=10, example=str(self.names))
```

**Comment by glenn-jocher** [comment]
Hi @Sscicada,

That's a good workaround! Modifying the `line_width` directly in the `Annotator` initialization within the `_run` function of the `Detections` class is an effective way to control the thickness of the bounding box lines. Just ensure that any changes you make align with your specific requirements for visualization.

Here's a slightly adjusted example to ensure clarity:

```python
if show or save or render or crop:
    annotator = Annotator(im, line_width=10, example=str(self.names))
```

This will set the line thickness for the bounding boxes to 10 pixels. You can adjust the `line_width` value as needed for your application. Thanks for sharing your solution! 😊 If you have any more questions or need further assistance, feel free to ask.
